contrib/src/templates/fairing.rs

```diff
@@ -0,0 +1,125 @@
+use super::DEFAULT_TEMPLATE_DIR;
+use super::context::Context;
+use super::engine::Engines;
+
+#[cfg(debug_assertions)]
+use super::watch::TemplateWatcher;
+
+use rocket::{Data, Request, Rocket};
+use rocket::config::ConfigError;
+use rocket::fairing::{Fairing, Info, Kind};
+
+#[cfg(not(debug_assertions))]
+mod context {
+    use std::ops::Deref;
+    use super::Context;
+
+    /// Wraps a Context. With `cfg(debug_assertions)` active, this structure
+    /// additionally provides a method to reload the context at runtime.
+    pub struct ContextManager(Context);
+
+    impl ContextManager {
+        pub fn new(ctxt: Context) -> ContextManager {
+            ContextManager(ctxt)
+        }
+
+        pub fn get<'a>(&'a self) -> impl Deref<Target=Context> + 'a {
+            &self.0
+        }
+    }
+}
+
+#[cfg(debug_assertions)]
+mod context {
+    use std::ops::{Deref, DerefMut};
+    use std::sync::RwLock;
+    use super::{Context, Engines, TemplateWatcher};
+
+    /// Wraps a Context. With `cfg(debug_assertions)` active, this structure
+    /// additionally provides a method to reload the context at runtime.
+    pub struct ContextManager{ context: RwLock<Context>, watcher: Option<TemplateWatcher> }
+
+    impl ContextManager {
+        pub fn new(ctxt: Context) -> ContextManager {
+            let root = ctxt.root.clone();
+            ContextManager {
```

Comment:
I think you can reorder this so that you don't need the `let` binding:

```rust
ContextManager {
    watcher: TemplateWather::new(ctxt.root.clone()),
    context: RwLock::new(ctxt)
}
```